rcore-fs-devfs/src/lib.rs

```diff
@@ -169,13 +169,22 @@ impl INode for DevRootINode {
         }
     }
 
-    fn get_entry(&self, id: usize) -> Result<String> {
+    fn get_entry(&self, id: usize) -> Result<(usize, FileType, String)> {
```

Comment:
Is this really needed? At the caller site of `get_entry`, one can still call `metadata` t get inode and FileType by calling `.metadata()`.

Reply:
Yes, but this require an additional lookup, maybe not optimal of time?

Reply:
> Is this really needed? At the caller site of `get_entry`, one can still call `metadata` to get inode and FileType by calling `.metadata()`.
Yes, but this requires an additional lookup by filename, maybe not optimal of time?

Reply:
> Is this really needed? At the caller site of `get_entry`, one can still call `metadata` to get inode and FileType by calling `.metadata()`.

Yes, but this requires an additional lookup by filename, maybe not optimal of time?

Reply:
This expansion aims at improving the implementation of sys_getdents64, see [here](https://github.com/EatenBagpipe/rCore/commit/0c4d53f8d41933a1c9b8ad11a4fc174d56b8d9fe)

Reply:
Maybe calling `.metadata` at https://github.com/EatenBagpipe/rCore/commit/0c4d53f8d41933a1c9b8ad11a4fc174d56b8d9fe#diff-e6ea9f693b986112cb877af98cd1babbR168 is better?

Reply:
Seems to make perfect sense... 全 部 白 写

Reply:
Maybe the previous commits can still be merged. Thanks a lot.

Reply:
I have got something wrong: the metadata of inode is dir inode itself, but not the files in it.

Reply:
Oh, I also see that... so maybe this is still meaningful?

Reply:
However, FileType can't be determined until reading the file inodes.

Reply:
It's hard to say whether it is a optimization.

Reply:
I would like to split `get_entry` into two calls: one with entry name, out with entry name and metadata.

Reply:
See https://github.com/rcore-os/rcore-fs/commit/09d31ee3ee52d7663ee87b65bdf6772b5edc339f#diff-b7746d34cb88af182346726db4ac0d33R686

Reply:
Seems much more convenient without too much modification. Thx.

Reply:
Maybe the default implementation of `get_entry_with_metadata` can be provided by calling `get_entry` and `get_metadata`, instead of return a result directly. This can save much implementation.

Reply:
I mean, `self.fs().get_inode().metadata()`, which requires `get_inode` to be added into the `FileSystem` trait